gnd/src/main/java/com/google/android/gnd/ui/projectselector/ProjectSelectorViewModel.java

```diff
@@ -15,48 +15,88 @@
  */
 package com.google.android.gnd.ui.projectselector;
 
+import android.util.Log;
+
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
+
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
+import com.google.android.gnd.rx.Result;
 import com.google.android.gnd.system.AuthenticationManager;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.vo.Project;
-import io.reactivex.Completable;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+
 import java.util.Collections;
 import java.util.List;
+
 import javax.inject.Inject;
 
 public class ProjectSelectorViewModel extends AbstractViewModel {
   private static final String TAG = ProjectSelectorViewModel.class.getSimpleName();
 
   private final DataRepository dataRepository;
   private final MutableLiveData<Resource<List<Project>>> projectSummaries;
-  private final AuthenticationManager authManager;
+  private final PublishSubject<Integer> projectSelections;
+  public final MutableLiveData<Project> activeProject;
+  public final MutableLiveData<Throwable> activateProjectErrors;
+  public final Observable<Result<Project>> activeProjectStream;
 
   @Inject
   ProjectSelectorViewModel(DataRepository dataRepository, AuthenticationManager authManager) {
     this.dataRepository = dataRepository;
     this.projectSummaries = new MutableLiveData<>();
-    this.authManager = authManager;
-  }
+    this.activeProject = new MutableLiveData<>();
+    this.activateProjectErrors = new MutableLiveData<>();
+    this.projectSelections = PublishSubject.create();
+
+    Observable<AuthenticationManager.User> user = authManager.getUser();
+
+    Observable<Result<Resource<List<Project>>>> availableProjects =
+        user.switchMap(Result.mapObservable(this.dataRepository::getProjectSummaries));
+
+    this.activeProjectStream =
+        projectSelections.switchMap(Result.mapObservable(this::selectActiveProject));
 
-  // TODO: Show message when no visible projects found.
-  public void loadProjectSummaries() {
-    // TODO(#24): Fix leaky subscriptions!
     disposeOnClear(
-        authManager
-            .getUser()
-            .flatMap(user -> dataRepository.getProjectSummaries(user))
-            .subscribe(summaries -> projectSummaries.setValue(summaries)));
+        activeProjectStream.subscribe(
+            Result.unwrap(activeProject::setValue, this::onActiveProjectError)));
+    disposeOnClear(
+        availableProjects.subscribe(
+            Result.unwrap(projectSummaries::setValue, this::onProjectSummariesError)));
```

Comment:
Since we're only going half-way to Rx town at this stage, I'm finding it hard to follow what's actually happening now. Let's put our heads together to make these a bit more readable once the fixes are merged.

Also, could you confirm that you've run and tested these changes locally and that the app is WAI?